test/ruby/test_gc.rb

```diff
@@ -117,6 +117,21 @@ def test_gc_config_disable_major_gc_start_always_works
     GC.start
   end
 
+  def test_gc_config_implementation
+    omit unless /darwin|linux/.match(RUBY_PLATFORM)
+
+    gc_name = (ENV['RUBY_GC_LIBRARY'] || "default")
+    assert_equal gc_name, GC.config[:implementation]
+  end
+
+  def test_gc_config_implementation_is_readonly
+    omit unless /darwin|linux/.match(RUBY_PLATFORM)
+
+    impl = GC.config[:implementation]
+    GC.config(implementation: "somethingelse")
```

Comment:
IMO we should raise if the user tries to change a read-only configuration at runtime.